src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -519,8 +519,8 @@ public static IAuthorityValidator CreateAuthorityValidator(AuthorityInfo authori
                         }
 
                         return updateEnvironment ?
-                                CreateAuthorityWithTenant(CreateAuthorityWithEnvironment(configAuthorityInfo, account.Environment).AuthorityInfo, account?.HomeAccountId?.TenantId) :
-                                CreateAuthorityWithTenant(configAuthorityInfo, account?.HomeAccountId?.TenantId);
+                                CreateAuthorityWithTenant(CreateAuthorityWithEnvironment(requestAuthorityInfo, account.Environment).AuthorityInfo, account?.HomeAccountId?.TenantId) :
```

Comment:
I don't think this is the correct fix.

Reply:
This gives precedence to request over configuration. We can discuss tomorrow.